libmariadb/mariadb_lib.c

```diff
@@ -1080,6 +1161,73 @@ int mthd_my_read_one_row(MYSQL *mysql,uint fields,MYSQL_ROW row, ulong *lengths)
   return 0;
 }
 
+/**
+ * If CLIENT_DEPRECATE_EOF isn't supported by the server, the response looked like
+ * <FieldCount><Metadata><EOF><ResultSet * n><EOF>.
+ *
+ * Otherwise, the response looks like <FieldCount><Metadata><ResultSet * n><EOF>.
+ * The number of metadata to look for is already available in FieldCount
+ * therefore first EOF is not necessary.
+ *
+ * Therefore this method is introduced to just read packet metadata 
+ * and is called by both mthd_my_read_query_result and mthd_stmt_read_prepare_response
+ */
+MYSQL_FIELD *mthd_my_read_metadata_ex(MYSQL *mysql, MA_MEM_ROOT *alloc,
```

Comment:
Similar to [mysql client](https://github.com/mysql/mysql-server/blob/bd9c260cf7026a54d9bf91ce6782cdb4e6a25c71/sql-common/client.c#L2179)